### ISSUE TITLE
Rework the footer

### ISSUE THREAD

**Issue by akeller** [camunda/camunda-platform-docs#492]
I hate the footer 🤣  and I can't always put my finger on why, but I think we could do a better job with adding a 4th column and thinking a bit more about why we put things there, what we put there, and when. 

If anyone has any thoughts, please chime in. Otherwise, I'll create a PR with my ideas and we can go from there.

**Comment by akeller** [comment]
@christinaausley @pepopowitz this is a good topic for both of you - you can flex your organizational skills and get a little more acquainted with how the footer works. I'll assign you both to discuss, collaborate, and determine how you want to move forward. When you have a branch or a PR ready for review, LMK!

Hint - you can add or remove most of the info here, you can also pull in other external links. 

Bonus points - In the internal Brand Assets we have a pop-out icon that should be added to external links. I've attached the white one.
![Camunda_Icon_Jan-2022_white_SVG_link-pop-out](https://user-images.githubusercontent.com/1051363/163252248-d369fd41-44e8-4c0d-be70-b1a6c87bd5aa.svg)

**Comment by akeller** [comment]
Ok that's hilarious but I promise it's an image and not a blank white square 😅  😆

**Comment by davidgs** [comment]
Was trying to get from the docs to the Forum, and of course looked in the footer, under `Community` because I thought "well, the "community forum" would be listed there." but instead, it is apparently listed instead on the `Contact us` page, which is where I would not have looked for it :-) 

![Screen Shot 2022-04-18 at 10 11 56 AM](https://user-images.githubusercontent.com/2071898/163821224-f147e186-5f3c-49f0-b33b-d21d3bdaa7a8.png)
 Maybe we could add that link to the Community Forum to the Community column?

**Comment by christinaausley** [comment]
@akeller I've just shared a Google doc with you outlining @pepopowitz and I's combined proposal for the footer/Reference section (based on our independent drafts for the footer). Looking forward to your thoughts, so we can revise and get to implementing.

**Comment by pepopowitz** [comment]
An update re: the external link icon: 

## TL;DR

Docusaurus includes an external link icon out of the box, but we're overriding the footer link component to not include it. 

A short term solution is to update our custom footer link to include this icon. But once we update to latest version of docusaurus, we'll need to rewrite our custom footer link component, because the original has changed in the last few versions. Because of this, I'm going to take a half day timebox to see what it will take to get docusaurus up to date. Once I reach my timebox, I'll decide whether it's worth continuing along the update path, or add the icon and deal with the update/footer link rewrite at a later date. 

## The journey, in excruciating detail

1. I noticed that docusaurus's website was displaying external link icons: 

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/1627089/167327908-61eb0e7a-772a-4590-bb6d-5e1242ad7ade.png">

That's pretty much what we want to do, so I thought we could look at their website for guidance.

2. I noticed in their source that they're not doing anything special to make the link appear: 

https://github.com/facebook/docusaurus/blob/main/website/docusaurus.config.js#L533-L536

It's just a link in their config, nothing specifying that there should be an icon showing. 

3. It looks like docusaurus adds that icon by default to any external link:

https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/Footer/LinkItem/index.tsx#L33

An external link is identified by the fact that there's a protocol specified in the URL: 

https://github.com/facebook/docusaurus/blob/main/packages/docusaurus/src/client/exports/isInternalUrl.ts#L12-L14

Our external links have the protocol defined, so it's surprising that we aren't already getting external link icons: 

https://github.com/camunda/camunda-platform-docs/blob/master/docusaurus.config.js#L123-L126

4. I thought maybe this was an issue of us being on an older version of docusaurus. 

I tracked down the footer link code in our version (2.0.0-beta.15). It appears that even that version should be including the external link icon: 

https://github.com/facebook/docusaurus/blob/v2.0.0-beta.15/packages/docusaurus-theme-classic/src/theme/Footer/index.tsx#L45-L50

At this point I started throwing breakpoints in my node_modules because I didn't understand why we were bypassing this. But then...

5. I noticed we're overriding the footer component! 

And in our override component, we're only emitting the link text, never an external link icon: 

https://github.com/camunda/camunda-platform-docs/blob/master/src/theme/Footer/index.js#L38

It looks like we override this component in order to add some analytics tracking. It's based on a specific version of the footer though, and I think one that's older than the 2.0.0-beta.15 version that we're using -- so it's _already_ out of date. 

6. I considered just adding the external link icon logic from docusaurus source into our override component. 

It seems like this should work fine. But I thought about how we also want to update the docusaurus dependency soon, and if we do, this footer code changes pretty significantly -- so we'd need to rewrite our override component once we update. I'd kind of like to only write this override logic once, so I'd like to see what it will take to do the update first, _then_ override the footer link to use our external icon instead of the docusaurus one.

<img src="https://media1.giphy.com/media/bOwOAey4MDO3ivBkgK/giphy.gif"/>

**Comment by christinaausley** [comment]
@pepopowitz This is such an interesting finding! Really enjoy the thought process you've stepped through so I can see (and learn) how you figured this out.

_Docusaurus includes an external link icon out of the box, but we're overriding the footer link component to not include it._

This makes sense. Are you comfortable updating the version, and rewriting the custom footer link component? It sounds like you're going to step through the update first to determine the best path forward and whether this is more worthwhile to pursue sooner versus later. Let me know if I can help assist/test in any way!

**Comment by pepopowitz** [comment]
Well that docusaurus update experiment ended pretty quickly 😅 

When updating the first interval, from beta-15 to beta-16, I ran into [a required upgrade on our Algolia account](https://docusaurus.io/blog/2021/11/21/algolia-docsearch-migration#upgrading-your-docusaurus-site). I don't feel comfortable making any decisions/movement on this until I talk to @akeller, as I know from conversations and [this Slack thread](https://camunda.slack.com/archives/GUJKW862Y/p1644449690519219) that she has lots of context on this update. 

Specifically, I can't move to beta-16 because it requires an algolia appID, which we acquire from Algolia when we update to their new DocSearch infrastructure.

I'm going to circle back and apply the external link changes to our current footer override. When we get back to updating docusaurus we'll also revisit this custom footer.
